sdk/src/main/java/io/radar/sdk/util/RadarSimpleLogBuffer.kt

```diff
@@ -3,59 +3,189 @@ package io.radar.sdk.util
+                // put back into disk
                 if (!success) {
-                    // Reverse order to ensure the logs will purge correctly (oldest logs purged first)
-                    logs.reverse()
-                    logs.forEach {
-                        if (!list.offerFirst(it)) {
-                            purgeOldestLogs()
+                    oldLogs.reverse()
+                    oldLogs.forEach {
+                        if (!oldList.offerFirst(it)) {
+                            oldPurgeOldestLogs()
                         }
                     }
+                    if (featureFlag) {
+                        purgeOldestLogs()
+                    }
                 }
             }
 
         }
+        
+    }
+
+
+
+    private fun oldPurgeOldestLogs() {
+        val oldLogs = mutableListOf<RadarLog>()
+        oldList.drainTo(oldLogs, HALF_CAPACITY)
+       if (!featureFlag) {
+            write(Radar.RadarLogLevel.DEBUG, "----- purged oldest logs -----", null)
+        } 
+        
     }
 
     /**
      * Clears oldest logs and adds a "purged" log line
      */
     private fun purgeOldestLogs() {
-        val logs = mutableListOf<RadarLog>()
-        list.drainTo(logs, HALF_CAPACITY)
-        write(Radar.RadarLogLevel.DEBUG, "----- purged oldest logs -----", null)
+        var files = RadarFileStorage(context).allFilesInDirectory(logFileDir)
+        if (files.isNullOrEmpty()) {
+            return 
+        }
+        var printedPurgedLogs = false
+        while(files?.size ?: 0 > MAXIMUM_CAPACITY){
+            val numberToPurge = min(HALF_CAPACITY,files?.size ?: 0)
+            for(i in 0 until numberToPurge){
+                files?.get(i)?.delete()
+            }
+            if(!printedPurgedLogs){
+                writeLogsToDisk(listOf(RadarLog(Radar.RadarLogLevel.DEBUG, "----- purged oldest logs -----", null)))
+                printedPurgedLogs = true
+            }
+            files = RadarFileStorage(context).allFilesInDirectory(logFileDir)
+        }
+    }
+
+    /**
+     * Gets logs from disk
+     */
+    private fun getLogsFromDisk(): LinkedBlockingDeque<RadarLog> {
+
+        val files = RadarFileStorage(context).allFilesInDirectory(logFileDir)
+        val logs = LinkedBlockingDeque<RadarLog>()
+        if (files.isNullOrEmpty()) {
+            return logs
+        }
+
+        for (file in files) {
+            val jsonString = RadarFileStorage(context).readFileAtPath(logFileDir, file.name)
+            val log = RadarLog.fromJson(JSONObject(jsonString))
+            if(log != null){
+                logs.add(log)
+            }
+        }
+        return logs
+    }
+
+    /**
+    * Writes logs to disk
+    */
+    private fun writeLogsToDisk(logs: Collection<RadarLog>) {
+        for (log in logs) {
+            val counterString = String.format("%04d", counter++)
+            val fileName = "${log.createdAt.time / 1000}${counterString}"
```

Comment:
might suggest putting an underscore or other separator between the timestamp and the counter for readability.

Reply:
I was originally appending the counter number at the back of each timestamp to "tie-break" logs created in the same millisecond. These filenames are converted to longs to be compared so as to get a chronological ordering of their timestamp. (We are not using the meta data of the file's creation time as we sometimes manually set the timestamp.) I agree with the point about readability, would it be worth it to add the `_` and use a more complicated comparison predicate func?